src/services/dynamo-database-service.ts

```diff
@@ -100,15 +100,24 @@ export class DynamoDatabaseService {
         ':ttl': { N: ttl.toString() },
         ':false': { BOOL: false },
       },
-      ConditionExpression: 'attribute_not_exists(isAccountDeleted) OR isAccountDeleted = :false',
+      ConditionExpression:
+        'attribute_exists(pk) AND (attribute_not_exists(isAccountDeleted) OR isAccountDeleted = :false)',
     };
     const command = new UpdateItemCommand(commandInput);
-    const response = await this.dynamoClient.send(command);
-    if (!response) {
-      const errorMessage = 'DynamoDB may have failed to update items, returned a null response.';
+    try {
+      const response = await this.dynamoClient.send(command);
+      logger.info(`${LOGS_PREFIX_SENSITIVE_INFO} Account ${userId} marked as deleted`);
+      logAndPublishMetric(MetricNames.MARK_AS_DELETED_SUCCEEDED);
+      if (!response) {
+        const errorMessage = 'DynamoDB may have failed to update items, returned a null response.';
+        logger.error(errorMessage);
+        logAndPublishMetric(MetricNames.DB_UPDATE_ERROR);
+      }
+      return response;
+    } catch {
+      const errorMessage = `Error updating item with pk ${userId}`;
```

Comment:
I think you should add the sensitive info prefix here since we are logging the user id